### PR TITLE
Feture/new command structure

### PR DESCRIPTION
**Pull request by CALEXCO**
New command structure to be able to create new project and build the project

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,112 +1,177 @@
...
-    project_name: String,
+    // project_name: Option<String>,
+    #[command(subcommand)]
+    command: Option<Commands>,
+}
+
+#[derive(Subcommand, Debug, Clone)]
+enum Commands {
+    /// Create new C Proejct with name
+    CNew {
```
Comment (pxp9) [inline src/main.rs]: Agree with clippy here, I think prefixes in Enum variants are redundant.

Please remove the `C` prefix for each variant.

The project itself is named `cinit` and in the description/Readme describes that it is creating a C language tool, so it is not needed to prefix with `c` in functions variants and whatever.

File: src/main.rs [hunk 2]
```diff
@@ -1,112 +1,176 @@
...
+    },
+
+    /// Build ./src/*.c files and moved to ./bin/
+    BuildC,
+    /// Build and run project
+    BulidCRunC,
 }
 
 #[derive(Clone)]
 struct Cinit(Args);
```
Comment (pxp9) [inline src/main.rs]: Remove Prefix plz

File: src/main.rs [hunk 3]
```diff
@@ -1,112 +1,176 @@
...
-        let cinit = Cinit(Args::parse());
-        let _ = fs::create_dir(cinit.clone().0.project_name);
-
-        cinit
+        Cinit(Args::parse())
     }
+}
+
+fn c_new(project_name: &String, lib: Option<String>) -> Result<(), Box<dyn std::error::Error>> {
+    create_sub_directory(project_name, "src")?;
```
Comment (pxp9) [inline src/main.rs:44]: Remove the prefix from the name of the function

File: src/main.rs [hunk 4]
```diff
@@ -1,112 +1,176 @@
...
+    Ok(())
+}
 
-    fn write_main_c(&self, file_dir: &str) {
-        let content = "#include <stdio.h>\n\nint main() {\n    printf(\"Hello, world!\\n\");\n    return 0;\n}\n";
-        let _ = fs::write(file_dir, content);
-        println!("Content written to {}", file_dir);
-    }
+fn build_c() -> Result<(), std::io::Error> {
+    // Ejecutar GCC para compilar el programa
```
Comment (pxp9) [inline src/main.rs:58]: Remove Prefix plz

File: src/main.rs [hunk 5]
```diff
@@ -1,112 +1,176 @@
...
+    let gcc_status = Command::new("gcc")
+        .args(["src/main.c", "-Wall", "-o", "main"])
+        .status()
+        .expect("Failed to execute GCC");
```
Comment (pxp9) [inline src/main.rs]: I think this command should be using the build system selected by the user.

If the user selects that wants a Makefile in the project I think this command should execute `make`

And you should consider the possibility of a different build system. For example `nob` , our dear friend tsoding is using it in their projects.

File: src/main.rs [hunk 6]
```diff
@@ -1,112 +1,176 @@
...
+    if let Some(command) = Cinit::new().0.command {
+        match command {
+            Commands::CNew { project_name, lib } => {
+                c_new(&project_name, lib)?;
+            }
+            Commands::BuildC => {
+                build_c()?;
+                //println!("{:?}", std::env::current_dir())
+            }
+            Commands::BulidCRunC => eprintln!("Developing"),
```
Comment (pxp9) [inline src/main.rs]: I think you can just use `todo!()` macro if this feature is not developed yet and the compiler will shut the fuck up

**Review by pxp9 — CHANGES_REQUESTED** [review]

File: src/main.rs (hunk 5 above)
Comment (CALEXCO) [reply src/main.rs]: For the moment the build command will only compile using gcc. Future versions will accept the build system selected by the user.

Comment (pxp9) [reply src/main.rs]: Put an issue and reference the issue in this conversation then

Comment (CALEXCO) [reply src/main.rs]: I have created the issue referring to your suggestion #6

**Pull request merged**